src/pybind/mgr/orchestrator/module.py

```diff
@@ -1256,6 +1260,40 @@ def _apply_snmp_gateway(self,
 
         return self._apply_misc([spec], dry_run, format, no_overwrite)
 
+    @_cli_write_command('orch apply jaeger')
+    def _apply_jaeger(self,
+                      es_nodes: Optional[str] = None,
+                      placement: Optional[str] = None,
```

Comment:
thanks for the review!

**jaeger-agent** - needs to be running on each host because it receives traces via UDP, so we want to avoid losses. for now, tracing works only on `RGW/OSD`, but in the future, I guess we would like to trace even more, so it won't be exclusive for rgw or osd.

**jaeger-collector/query** - you are right. we should deploy as default 1 query. 
the collectors number should be configurable (according to cluster size, each agent sends its data to the collector(s) )


in case we'd like a default deployment, we can deploy single elasticsearch, and single jaeger-query. that's the minimal requirement for jaeger.
in case of customized deployment, we can allow an option for self-deployment of elasticsearch (on specific host, specific setup), and also allow self-deployment of jaeger-query. there are a lot of options and confs and it should be out of the scope of this PR.  
in both cases that I mentioned, jaeger-agent and jaeger-collector will be deployed by cephadm. (num of collectors can be configurable for scalability)

Reply:
alright, I'm fine with leaving the extra placement configuration for a future PR. Was just curious if there was a use case for it and it sounds like there is in terms of different customizable placements for the collector, query and elasticsearch.

Reply:
if you aren't going to do it in this PR though would you mind creating a tracker for doing so when we merge this so it doesn't get forgotten?

Reply:
added it in [6cd9cf7](https://github.com/ceph/ceph/pull/44493/commits/6cd9cf79274b4e999488612b7bdbde1fdcd2d4a7)
it will allow us to deploy the `collector` and `agent`, and leave the `query` and `es` for self-deployment. (the elasticsearch addresses need to be passed to the apply command)